blockchain/chain.go

```diff
@@ -883,7 +912,7 @@ func (b *BlockChain) reorganizeChain(detachNodes, attachNodes *list.List) error
 
 		// Load all of the utxos referenced by the block that aren't
 		// already in the view.
-		err = view.fetchInputUtxos(b.db, block)
+		err = view.addInputUtxos(b.utxoCache, block, true)
```

Comment:
Does this need CTOR detection to decide if it can ignore out of order or not?

Reply:
So since these blocks have already been validated we aren't validating the ordering here, only updating the utxos (that's the idea at least unless I'm missing something).